models/running_input.go

```diff
@@ -114,6 +117,16 @@ func (r *RunningInput) Init() error {
 		return fmt.Errorf("invalid 'startup_error_behavior' setting %q", r.Config.StartupErrorBehavior)
 	}
 
+	if _, ok := r.Input.(telegraf.ServiceInput); !ok {
```

Comment:
Please don't do this! Some plugins implement both the `ServiceInput` interface as well as a `Gather` method for being able to establishing connections during `Start()`. Currently there is no reliable way to judge if a plugin is a pure service plugin or not.
Long story short: Always set the time-source and always override the time in `Gather`! Service inputs simply don't return anything there.